src/Api/Mastodon/Account.php

```diff
@@ -52,37 +53,52 @@ class Account
 	var $fields = null;
 	/** @var bool|null */
 	var $bot = null;
+	/** @var bool */
+	var $group;
+	/** @var bool */
+	var $discoverable;
+	/** @var string|null (Datetime) */
+	var $last_status_at = null;
 
 	/**
-	 * Creates an account record from a contact record. Expects all contact table fields to be set
+	 * Creates an account record from a public contact record. Expects all contact table fields to be set.
 	 *
-	 * @param array $contact   Full contact table record
-	 * @param array $apcontact Full apcontact table record
+	 * @param BaseURL $baseUrl
+	 * @param array   $publicContact Full contact table record with uid = 0
+	 * @param array   $apcontact     Optional full apcontact table record
 	 * @return Account
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function createFromContact(array $contact, array $apcontact = [])
+	public static function create(BaseURL $baseUrl, array $publicContact, array $apcontact = [])
 	{
 		$account = new Account();
-		$account->id              = $contact['id'];
-		$account->username        = $contact['nick'];
-		$account->acct            = $contact['nick'];
-		$account->display_name    = $contact['name'];
+		$account->id              = $publicContact['id'];
+		$account->username        = $publicContact['nick'];
+		$account->acct            =
+			strpos($publicContact['url'], $baseUrl->get() . '/') === 0 ?
+			$publicContact['nick'] :
+			$publicContact['addr'];
+		$account->display_name    = $publicContact['name'];
 		$account->locked          = !empty($apcontact['manually-approve']);
-		$account->created_at      = DateTimeFormat::utc($contact['created'], DateTimeFormat::ATOM);
+		$account->created_at      = DateTimeFormat::utc($publicContact['created'], DateTimeFormat::ATOM);
 		$account->followers_count = $apcontact['followers_count'] ?? 0;
 		$account->following_count = $apcontact['following_count'] ?? 0;
 		$account->statuses_count  = $apcontact['statuses_count'] ?? 0;
-		$account->note            = BBCode::convert($contact['about'], false);
-		$account->url             = $contact['url'];
-		$account->avatar          = $contact['avatar'];
-		$account->avatar_static   = $contact['avatar'];
+		$account->note            = BBCode::convert($publicContact['about'], false);
+		$account->url             = $publicContact['url'];
+		$account->avatar          = $publicContact['avatar'];
```

Comment:
For feeds and mail contacts we should use the "contact" avatar.

Reply:
What does this mean?

Reply:
If the network is "mail" or "feed" we should take the avatar picture from the personal contact entry. For these networks the avatar isn't overwritten automatically like with all other networks.

Reply:
Do you mean “updated”?

Reply:
Yes.